x-pack/auditbeat/module/system/packages/packages.go

```diff
@@ -0,0 +1,356 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package packages
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/elastic/beats/x-pack/auditbeat/cache"
+	"github.com/elastic/beats/x-pack/auditbeat/module/system/config"
+
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/go-sysinfo"
+)
+
+const (
+	moduleName    = "system"
+	metricsetName = "packages"
+)
+
+func init() {
+	mb.Registry.MustAddMetricSet(moduleName, metricsetName, New,
+		mb.DefaultMetricSet(),
+	)
+}
+
+// MetricSet collects data about the host.
+type MetricSet struct {
+	mb.BaseMetricSet
+	config config.Config
+	cache  *cache.Cache
+	log    *logp.Logger
+}
+
+// Package represents information for a package.
+type Package struct {
+	Name        string
+	Version     string
+	Release     string
+	Arch        string
+	License     string
+	InstallTime time.Time
+	Size        uint64
+	Summary     string
+	URL         string
+}
+
+// Hash creates a hash for Package.
+func (pkg Package) Hash() string {
+	// Could use real hash e.g. FNV if there is an advantage
+	return pkg.Name + pkg.InstallTime.String()
```

Comment:
I've switched to xxhash. And in `DiffAndUpdateCache` I've added logic at the beginning that stores the results of `Hash()` so that it's only ever called once in the lifetime of an object.